internal/api/handler.go

```diff
@@ -282,21 +282,40 @@ func (h *Handler) handleTokenExchange(c *OAuthAppContext, w http.ResponseWriter,
 		return
 	}
 
-	code := r.Form.Get("code")
-	if len(code) == 0 {
-		fmt.Println("Getting code from query")
-		code = r.URL.Query().Get("code")
-	}
-	if len(code) == 0 {
-		c.Logger.Error("Missing authorization code")
-		http.Error(w, "missing authorization code", http.StatusBadRequest)
-		return
-	}
+	grantType := r.Form.Get("grant_type")
```

Comment:
I think we should check if `len(grantType)` is 0 and handle that error as we did for `code`.